linux_os/guide/system/software/sudo/sudo_require_reauthentication/bash/shared.sh

```diff
@@ -7,20 +7,20 @@
 
 {{{ bash_instantiate_variables("var_sudo_timestamp_timeout") }}}
 
-if grep -x '^[\s]*Defaults.*\btimestamp_timeout=.*' /etc/sudoers.d/*; then
-    find /etc/sudoers.d/ -type f -exec sed -i "/^[\s]*Defaults.*\btimestamp_timeout=.*/d" {} \;
+if grep -Px '^[\s]*Defaults.*timestamp_timeout[\s]*=.*' /etc/sudoers.d/*; then
+    find /etc/sudoers.d/ -type f -exec sed -Ei "/^[[:blank:]]*Defaults.*timestamp_timeout[[:blank:]]*=.*/d" {} \;
 fi
 
 if /usr/sbin/visudo -qcf /etc/sudoers; then
     cp /etc/sudoers /etc/sudoers.bak
-    if ! grep -P '^[\s]*Defaults.*\btimestamp_timeout=[-]?\w+\b\b.*$' /etc/sudoers; then
+    if ! grep -P '^[\s]*Defaults.*timestamp_timeout[\s]*=[\s]*[-]?\w+.*$' /etc/sudoers; then
         # sudoers file doesn't define Option timestamp_timeout
         echo "Defaults timestamp_timeout=${var_sudo_timestamp_timeout}" >> /etc/sudoers
     else
         # sudoers file defines Option timestamp_timeout, remediate if appropriate value is not set
-        if ! grep -P "^[\s]*Defaults.*\btimestamp_timeout=${var_sudo_timestamp_timeout}\b.*$" /etc/sudoers; then
+        if ! grep -P "^[\s]*Defaults.*timestamp_timeout[\s]*=[\s]*${var_sudo_timestamp_timeout}.*$" /etc/sudoers; then
             
-            sed -Ei "s/(^[\s]*Defaults.*\btimestamp_timeout=)[-]?\w+(\b.*$)/\1${var_sudo_timestamp_timeout}\2/" /etc/sudoers
+            sed -Ei "s/(^[[:blank:]]*Defaults.*timestamp_timeout[[:blank:]]*=)[[:blank:]]*[-]?\w+(.*$)/\1${var_sudo_timestamp_timeout}\2/" /etc/sudoers
```

Comment:
Was the change to `[[:blank:]]` to improve the group capturing?

I'm being a bit nitpicky, but could you add in the commit message the reasoning?

Reply:
I changed it because (at least on my testing environments) `sed` seems to not recognize the `\s` syntax and the bash remediation was not working properly:
```
[root@localhost ~]# cat /etc/sudoers.d/00-complianceascode-test.conf 
Defaults timestamp_timeout = 3
Defaults timestamp_timeout= 6
Defaults timestamp_timeout=6
Defaults timestamp_timeout =6
  Defaults timestamp_timeout=6
[root@localhost ~]# sed -E '/^[\s]*Defaults.*timestamp_timeout[\s]*=.*/d' /etc/sudoers.d/00-complianceascode-test.conf
Defaults timestamp_timeout = 3
Defaults timestamp_timeout =6
  Defaults timestamp_timeout=6
[root@localhost ~]# sed -E '/^[[:blank:]]*Defaults.*timestamp_timeout[[:blank:]]*=.*/d' /etc/sudoers.d/00-complianceascode-test.conf
[root@localhost ~]# 
```